packages/flutter_tools/lib/src/windows/application_package.dart

```diff
@@ -20,12 +21,50 @@ abstract class WindowsApp extends ApplicationPackage {
     );
   }
 
-  /// Creates a new [WindowsApp] from an existing executable.
+  /// Creates a new [WindowsApp] from an existing executable or a zip archive.
   ///
-  /// `applicationBinary` is the path to the executable.
-  factory WindowsApp.fromPrebuiltApp(FileSystemEntity applicationBinary) {
+  /// `applicationBinary` is the path to the executable or the zipped archive.
+  static WindowsApp? fromPrebuiltApp(FileSystemEntity applicationBinary) {
+    if (applicationBinary.path.endsWith('.exe')) {
+      return PrebuiltWindowsApp(
+        executable: applicationBinary.path,
+        applicationPackage: applicationBinary,
+      );
+    }
+
+    if (!applicationBinary.existsSync()) {
+      globals.printError('File "${applicationBinary.path}" does not exist.');
+      return null;
+    }
+
+    // Try to unpack as a zip.
+    final Directory tempDir = globals.fs.systemTempDirectory.createTempSync('flutter_app.');
+    try {
+      globals.os.unzip(globals.fs.file(applicationBinary), tempDir);
+    } on ProcessException {
```

Comment:
I believe on Windows this uses `package:archive`, so i doubt this will throw a `ProcessException`. Looks like hte code throws `ArchiveException`